src/stage/approved.rs

```diff
@@ -1449,48 +1386,65 @@ impl ApprovedPeer {
         }
     }
 
-    fn vote_for_event(&mut self, event: AccumulatingEvent) {
-        self.vote_for_network_event(event.into_network_event())
-    }
-
-    fn vote_for_network_event(&mut self, event: NetworkEvent) {
-        trace!("Vote for Event {:?}", event);
-        self.parsec_map.vote_for(event);
-    }
-
     ////////////////////////////////////////////////////////////////////////////
     // Message sending
     ////////////////////////////////////////////////////////////////////////////
 
-    fn send_neighbour_infos(&mut self) {
+    // Send NodeApproval to the current candidate which makes them a section member and allows them
+    // to passively participate in parsec consensus (that is, they can receive gossip and poll
+    // consensused blocks out of parsec, but they can't vote yet)
+    fn send_node_approval(
+        &mut self,
+        core: &mut Core,
+        p2p_node: P2pNode,
+        their_knowledge: Option<u64>,
+    ) {
+        info!(
+            "Our section with {:?} has approved candidate {}.",
+            self.chain.our_prefix(),
+            p2p_node
+        );
+
+        let trimmed_info = self.gen_pfx_info.trimmed();
+        let src = SrcLocation::Section(*trimmed_info.elders_info.prefix());
+        let dst = DstLocation::Node(*p2p_node.name());
+
+        let variant = Variant::NodeApproval(Box::new(trimmed_info));
+        if let Err(error) = self.send_routing_message(core, src, dst, variant, their_knowledge) {
```

Comment:
**(General discussion, for eventual consideration.)**

For testability I would prefer to not have the sending within this object. Instead this would be a side-effect free method producing a cmd. Which by upper layer can be dispatched to a single place of sender logic.
It seems like the logic here supports that idea as we don't really care if it succeeded or not (no handling of error, just a debug msg).

(It's one of the things we would achieve by being explicit with the Saga pattern btw.)

Reply:
Completely agree, and it's something I wanted to do, but it's somehow tricky in practice. I have some ideas but need to spend some time on it to find the best approach.